### ISSUE TITLE
Handle single character command line arguments correctly

### ISSUE THREAD

**Issue by redcatbear** [hamstercommunity/openfasttrack#25]
# Situation

As per GNU convention the following is equivalent:

```
myprogram --verbose --recursive
myprogram -verbose -recursive
myprogram -v -r
myprogram -vr
```
# Acceptance criteria
1. Multi-character argument names are led in by double dash or single dash
2. Single-character argument names are led in by single dash
3. Single-character argument names for arguments without values can be chained after a single dash
4. Multi-word argument names are lowercase and separated with dash (e.g. `--output-dir`)
# See Also

http://www.gnu.org/software/libc/manual/html_node/Getopt-Long-Options.html

**Comment by redcatbear** [comment]
Please close after review.

**Comment by kaklakariada** [comment]
Closed with #42
